### PR TITLE
Homology ref species

### PR DESCRIPTION
**Pull request by kamaldodiya**
## Description

Add homology reference_species to stats endpoint.

## Related JIRA
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2317

### Example 1 - reference_species
```
curl  'http://localhost:8014/api/metadata/genome/1cda6f4c-8f49-4305-b435-5cbe00adec9a/stats'  | jq '.["genome_stats"]["homology_stats"]' 
{
  "coverage": 40.7,
  "reference_species_name": "Triticum turgidum subsp. durum"
}
```

### Example 2 - reference_species
```
curl 'http://localhost:8014/api/metadata/genome/a7335667-93e7-11ec-a39d-005056b38ce3/stats' | jq '.["genome_stats"]["homology_stats"]'    
{
  "coverage": 91.6,
  "reference_species_name": "Pan troglodytes"
}

```
### Example 3 - null
(Spoofed the stats in order to produce this)
```
curl "http://localhost:8014/api/metadata/genome/${gid}/stats" | jq '.["genome_stats"]["homology_stats"]'
{
  "coverage": null,
  "reference_species_name": null
}


```

### CODE REVIEW

File: app/api/models/statistics.py [hunk 1]
```diff
@@ -7,9 +7,21 @@
 from loguru import logger
 
 class Homology(BaseModel):
-    coverage: float = Field(alias="homology_coverage", default=None)
-    coverage_explanation: str = None
-
+    coverage: float = Field(alias="compara.homology_coverage ", default=None)
+    reference_species_name: str = Field(alias="compara.homology_reference_species ", default=None)
```
Comment (azangru) [inline app/api/models/statistics.py:11]: Does the extra space before the closing double quote matter?

**Review by veidenberg — APPROVED** [review]

File: app/api/models/statistics.py [hunk 2]
```diff
@@ -7,9 +7,21 @@
...
+
+    @validator('reference_species_name', pre=True)
+    def validate_na_values(cls, v ) -> str:
+        rsn = ""
```
Comment (Mehrnaz-Charkhchi) [inline app/api/models/statistics.py:15]: What is rsn? Can we please use full names like we do everywhere in the code?

Comment (kamaldodiya) [reply app/api/models/statistics.py]: Done

File: app/api/models/statistics.py (hunk 1 above)
Comment (kamaldodiya) [reply app/api/models/statistics.py]: Updated code to handle both once production fixes this

**Comment by azangru** [comment]
```
curl 'http://localhost:8014/api/metadata/genome/82c2890-5c50-47f1-bbce-ad8daee14861/stats' | jq '.["genome_stats"]["homology_stats"]'
null
```

In your curl example, why is `["homology_stats"]` `null` rather than `{coverage: null, reference_species_name: null}`?

**Comment by kamaldodiya** [comment]
> ```
> curl 'http://localhost:8014/api/metadata/genome/82c2890-5c50-47f1-bbce-ad8daee14861/stats' | jq '.["genome_stats"]["homology_stats"]'
> null
> ```
> 
> In your curl example, why is `["homology_stats"]` `null` rather than `{coverage: null, reference_species_name: null}`?

My bad, That's because that's the organism_id not genome_uuid ;) 
```
curl 'https://staging-2020.ensembl.org/api/metadata/genome/825d7e53-8c34-42a7-83b2-196265151e88/stats' | jq '.["genome_stats"]["homology_stats"]'

{
  "coverage": null,
  "coverage_explanation": null
}
```

**Review by azangru — APPROVED** [review]

**Comment by azangru** [comment]
> This is on staging though . Do you have any specific example for genome_uuid which I can try ?

On staging? I am seeing:

image

**Review by Mehrnaz-Charkhchi — APPROVED** [review]

**Pull request merged**